modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1142,10 +1175,23 @@ def _get_dict_of_block_index(self, axis, indices):
                         )
                         dict_of_slices.update({last_part: slice(None, last_idx[0])})
                         return dict_of_slices
-        # Sort and convert negative indices to positive
-        indices = np.sort(
-            [i if i >= 0 else max(0, len(self.axes[axis]) + i) for i in indices]
-        )
+        if isinstance(indices, list):
+            # Converting python list to numpy for faster processing
+            indices = np.array(indices, dtype=np.int64)
+        negative_mask = np.less(indices, 0)
+        has_negative = np.any(negative_mask)
+        if has_negative:
+            # We're going to modify 'indices' inplace in a numpy way, so doing a copy/converting indices to numpy.
+            indices = (
+                indices.copy()
```

Comment:
we modify the passed `indices` inplace, so all the changes propagate to the user's object, we're doing a copy to avoid this
```python
>>> indices = np.array([1, 2, -1, 3])
>>> _ = get_dict_indices_with_copy(indices)
>>> indices
array([ 1,  2, -1,  3])
>>> _ = get_dict_indices_no_copy(indices)
>>> indices
array([1, 2, 4, 3])
```

Reply:
If indices are used further, then yes, copy makes sense.